### PR TITLE
[PROTOCOL-87] Integration Tests: Create Repayment Scenarios I

### PR DESCRIPTION
**Pull request by sypeer**
It is:
- Integration tests to cover the loan repayments.

It includes:
- Two integration tests:
-- Repay in 3 times with loan terms: TokenName/ETH ; 30 days; 1000 TokenName
-- Repay in 4 times with loan terms: TokenName/LINK ; 60 days; 800 TokenName

Screen Shot 2020-07-29 at 4 01 34 PM

### CODE REVIEW

File: test-integration/repay-loan/1_repay_loan_3_payments.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  await timer.advanceBlockAtTime(thirdPaymentTimestamp);
+
+  // Make 3rd payment
+  console.log('Making final payment...');
+  await token.approve(lendingPoolInstance.address, payment, borrowerTxConfig);
+  const repay3_result = await loansInstance.repay(payment, lastLoanID, borrowerTxConfig);
+
+  // Check final loan status 
+  totalOwedResult = totalOwedResult.minus(payment);
+  console.log('Expected (after 3# repay) total owed:  ', totalOwedResult.toString());
```
Comment (salazarguille) [inline test-integration/repay-loan/1_repay_loan_3_payments.js:180]: Please, remove this comment

File: test-integration/repay-loan/1_repay_loan_3_payments.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  loans
+    .loanRepaid(repay3_result)
+    .emitted(lastLoanID, borrowerTxConfig.from, payment, borrowerTxConfig.from, totalOwedResult.toString());
+  const loanStatus = await loansInstance.loans(lastLoanID);
```
Comment (salazarguille) [inline test-integration/repay-loan/1_repay_loan_3_payments.js:184]: Please, add assertions for the loanStatus.principalOwed == 0 and loanStatus.interestOwed == 0

File: test-integration/repay-loan/2_repay_loan_4_payments.js [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+  await timer.advanceBlockAtTime(forthPaymentTimestamp);
+
+  // Make 4th payment
+  console.log('Making final payment...');
+  await token.approve(lendingPoolInstance.address, payment, {from:borrower});
+  const repay4_result = await loansInstance.repay(payment, lastLoanID, borrowerTxConfig);
+
+  // Check final loan status 
+  totalOwedResult = totalOwedResult.minus(payment);
+  console.log('Expected (after 4# repay) total owed:  ', totalOwedResult.toString());
```
Comment (salazarguille) [inline test-integration/repay-loan/2_repay_loan_4_payments.js:215]: Please, remove this comment

File: test-integration/repay-loan/2_repay_loan_4_payments.js [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+  loans
+    .loanRepaid(repay4_result)
+    .emitted(lastLoanID, borrowerTxConfig.from, payment, borrowerTxConfig.from, totalOwedResult.toString());
+  const loanStatus = await loansInstance.loans(lastLoanID);
```
Comment (salazarguille) [inline test-integration/repay-loan/2_repay_loan_4_payments.js:219]: Please, add assertions for the loanStatus.principalOwed == 0 and loanStatus.interestOwed == 0

File: test-integration/repay-loan/1_repay_loan_3_payments.js [hunk 5]
```diff
@@ -0,0 +1,190 @@
+// Util classes
```
Comment (salazarguille) [inline test-integration/repay-loan/1_repay_loan_3_payments.js:1]: Please, remove filename to 1_repay_loan_eth_3_payments.js (or similar). So we know it uses ETH as collateral.

File: test-integration/repay-loan/2_repay_loan_4_payments.js [hunk 6]
```diff
@@ -0,0 +1,225 @@
+// Util classes
```
Comment (salazarguille) [inline test-integration/repay-loan/2_repay_loan_4_payments.js:1]: Please, remove filename to 1_repay_loan_eth_4_payments.js (or similar). So we know it uses ETH as collateral.

File: test-integration/repay-loan/1_repay_loan_3_payments.js (hunk 5 above)
Comment (sypeer) [reply test-integration/repay-loan/1_repay_loan_3_payments.js:1]: renamed

File: test-integration/repay-loan/2_repay_loan_4_payments.js (hunk 6 above)
Comment (sypeer) [reply test-integration/repay-loan/2_repay_loan_4_payments.js:1]: renamed

File: test-integration/repay-loan/1_repay_loan_3_payments.js (hunk 1 above)
Comment (sypeer) [reply test-integration/repay-loan/1_repay_loan_3_payments.js:180]: Removed

File: test-integration/repay-loan/2_repay_loan_4_payments.js (hunk 3 above)
Comment (sypeer) [reply test-integration/repay-loan/2_repay_loan_4_payments.js:215]: Removed

File: test-integration/repay-loan/1_repay_loan_3_payments.js (hunk 2 above)
Comment (sypeer) [reply test-integration/repay-loan/1_repay_loan_3_payments.js:184]: Added

File: test-integration/repay-loan/2_repay_loan_4_payments.js (hunk 4 above)
Comment (sypeer) [reply test-integration/repay-loan/2_repay_loan_4_payments.js:219]: Added

**Comment by sypeer** [comment]
> Good job! Syed I added some comments.

Thank you! Fixed as per your comments

File: test-integration/repay-loan/2_repay_loan_eth_4_payments.js [hunk 7]
```diff
@@ -0,0 +1,232 @@
+// Util classes
```
Comment (Juan-cc) [inline test-integration/repay-loan/2_repay_loan_eth_4_payments.js:1]: As this integration tests refers to LINK token, I think it'd be better if we also mention that in the filename as follows:
2_repay_loan_link_4_payments.js

File: test-integration/repay-loan/index.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+const _1_repay_loan_eth_3_payments = require('./1_repay_loan_eth_3_payments');
+const _2_repay_loan_eth_4_payments = require('./2_repay_loan_eth_4_payments');
+
+module.exports = {
+    'repay-loan-1-repay-loan-3-payments': _1_repay_loan_eth_3_payments,
+    'repay-loan-2-repay-loan-4-payments': _2_repay_loan_eth_4_payments,
```
Comment (Juan-cc) [inline test-integration/repay-loan/index.js:6]: Change 2_repay_loan_eth_4_payments.js with 2_repay_loan_link_4_payments.js

Comment (sypeer) [reply test-integration/repay-loan/index.js:6]: Fixed

File: test-integration/repay-loan/2_repay_loan_eth_4_payments.js (hunk 7 above)
Comment (sypeer) [reply test-integration/repay-loan/2_repay_loan_eth_4_payments.js:1]: Good point, changed

**Pull request merged**